src/Datadog.Trace.ServiceFabric/Remoting.cs

```diff
@@ -216,14 +216,14 @@ private static void GetMessageHeaders(EventArgs? eventArgs, out ServiceRemotingR
 
         private static void InjectContext(PropagationContext context, IServiceRemotingRequestMessageHeader messageHeaders)
         {
-            if (context.TraceId == 0 || context.ParentSpanId == 0)
+            if (context.TraceId.Equals(TraceId.Zero) || context.ParentSpanId == 0)
             {
                 return;
             }
 
             try
             {
-                messageHeaders.TryAddHeader(HttpHeaderNames.TraceId, context, ctx => BitConverter.GetBytes(ctx.TraceId));
+                messageHeaders.TryAddHeader(HttpHeaderNames.TraceId, context, ctx => ConvertHexStringToByteArray(ctx.TraceId.ToString()));
```

Comment:
I feel little worried about the performance of multiple conversions: `lower+upper ulongs` -> `hex string` -> `byte[]`

Reply:
That may save some time but it may also have an impact on memory if that TraceId is copied around (and then triggers creating a new string each time). At the same time, if we're only calling ToString right before a network operation happens, the difference in impact is probably small.